mlir/lib/Dialect/Rock/utility/transformMapUtils.cpp

```diff
@@ -502,7 +498,7 @@ ContiguousMergesMap findContiguousGroups(ArrayAttr transforms,
         auto fasterParam = std::get<0>(data[0]);
         if (maybeUnmerge && fasterParam == 1 && upperBounds.size()) {
           uint32_t slowerDim = std::get<1>(data.back());
-          unmergeParams.push_back(upperBounds[0]);
+          unmergeParams.push_back(upperBounds[upperBoundIdx]);
```

Comment:
... Oh. I'll note that `upperBounds` is global. You want to be looking up `upperBounds[embed.getUpperDims()[slowerDim]`, if I understand your code correctly.

Test case:
- `(x) -> (x / 4, x % 4)`
- `(a, b) -> (b, a)` `// (x % 4, x / 4)`
- `<Embed{1, 4} ["a", "b"] at [0, 1] -> ["y"] at [0]>` aka `(a, b) -> (1 * a + 4 * b)`

Reply:
I see few issues with `upperBounds[embed.getUpperDims()[slowerDim]` (correct me if I am wrong) : 
- `Slowerdim` is not a position and `getUpperDims` is a vector, not a map. If I have `getUpperDims=[3,4,2,1]` and `slowerDIm` is 4, this will get out of bound
- You might have some dimensions that have been deleted by `AddDim` and then you can easily get out of bounds.

Reply:
The point is that
1. The Embed takes dimension indexes (which map exactly to positions in `upperBounds`) as the upper dimensions
2. If you've sorted `zip(params, upperDims())`, then that second thing will be valid an index into `upperBounds`

Note that `upperBounds` is an attribute that exists on each transform_map.

Reply:
Yes, now I see what you mean. I amended the code and added the test